Composer/packages/tools/language-servers/language-understanding/src/LUServer.ts

```diff
@@ -243,6 +245,38 @@ export class LUServer {
     this.LUDocuments.push(luDocument);
   }
 
+  protected async findAllImportedEntities(
+    imports: { id: string; path: string; description: string }[],
+    importResolver: (source: string, id: string) => Promise<{ id: string; content: any }>,
+    luFeatures: any
+  ): Promise<string[]> {
+    let content = '';
+    const result: string[] = [];
+    for (const importFile of imports) {
+      try {
+        content = (await importResolver('.', importFile.id)).content;
+      } catch (error) {
+        // ignore if file not exist
+      }
+
+      const { resource } = await this.luParser.parse(content, importFile.id, luFeatures);
+      const sections = resource.Sections;
+      for (const section of sections) {
+        if (section.SectionType === 'newEntitySection') {
+          result.push(section.Name);
+        } else if (section.SectionType === 'simpleIntentSection') {
+          for (const entity of section.Entities) {
+            if (entity.SectionType === 'newEntitySection') {
+              result.push(entity.Name);
+            }
+          }
+        }
+      }
+    }
+
+    return Promise.resolve(uniq(result));
```

Comment:
>Promise.resolve [](http://example.com/codeflow?start=11&length=15)

minor question: is `Promise.resolve` needed here?

Reply:
removed